python/ray/serve/_private/long_poll.py

```diff
@@ -247,24 +249,29 @@ async def listen_for_change(
         immediately if the snapshot_ids are outdated, otherwise it will block
         until there's an update.
         """
-        watched_keys = keys_to_snapshot_ids.keys()
-        existent_keys = set(watched_keys).intersection(set(self.snapshot_ids.keys()))
-
         # If there are any keys with outdated snapshot ids,
         # return their updated values immediately.
-        updated_objects = {
-            key: UpdatedObject(self.object_snapshots[key], self.snapshot_ids[key])
-            for key in existent_keys
-            if self.snapshot_ids[key] != keys_to_snapshot_ids[key]
-        }
+        updated_objects = {}
+        for key, snapshot_id in keys_to_snapshot_ids.items():
+            try:
+                existing_id = self.snapshot_ids[key]
+            except KeyError:
+                # The caller may ask for keys that we don't know about (yet),
+                # just ignore them.
+                continue
+
+            if existing_id != snapshot_id:
+                updated_objects[key] = UpdatedObject(
+                    self.object_snapshots[key], existing_id
+                )
```

Comment:
We can avoid raising and catching an error by using a conditional block.

```suggestion
            # The caller may ask for keys that we don't know about (yet),
            # just ignore them.
            if key in self.snapshot_ids:
                latest_id = self.snapshot_ids[key]
                if latest_id != snapshot_id:
                    updated_objects[key] = UpdatedObject(
                        self.object_snapshots[key], latest_id
                    )
```